main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -1389,16 +1389,24 @@ object Weeder2 {
                 case Pattern.Wild(loc) => Validation.success(WeededAst.RestrictableChoosePattern.Wild(loc))
                 case Pattern.Var(ident, loc) => Validation.success(WeededAst.RestrictableChoosePattern.Var(ident, loc))
                 case Pattern.Cst(Ast.Constant.Unit, loc) => Validation.success(WeededAst.RestrictableChoosePattern.Wild(loc))
-                case other => Validation.toHardFailure(UnsupportedRestrictedChoicePattern(isStar, other.loc))
+                case other =>
+                  val e = UnsupportedRestrictedChoicePattern(isStar, other.loc)
+                  Validation.toSoftFailure(WeededAst.RestrictableChoosePattern.Error( other.loc.asSynthetic), e)
               }
-            case other => Validation.toHardFailure(UnsupportedRestrictedChoicePattern(isStar, other.loc))
+            case other =>
+              val e = UnsupportedRestrictedChoicePattern(isStar, other.loc)
+              Validation.toSoftFailure(List(WeededAst.RestrictableChoosePattern.Error(other.loc.asSynthetic)), e)
           }
           mapN(inner)(RestrictableChoosePattern.Tag(qname, _, loc))
-        case other => Validation.toHardFailure(UnsupportedRestrictedChoicePattern(isStar, other.loc))
+        case other =>
```

Comment:
@mlutze here

Reply:
I'm confused. Why do we construct a tag?

Reply:
Before this we returned a `HardFailure` but that requires `UnsupportedRestrictedChoicePattern` to be `Unrecoverable`. Now that we introduce `SoftFailure`s we require that the error i `Recoverable`. This means we cannot return the `HardFailure` like before so we are forced to return a value. Returning an error node i what https://github.com/flix/flix/pull/8586 does. Note however, that this returns a `Tag` with and `Error` inside. I think we should combine the two PRs (by first merging this).

Reply:
Ah, so you'll undo all the implicit flixes?

Reply:
Yes :)